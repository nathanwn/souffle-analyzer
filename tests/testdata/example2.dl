/// There exists an edge between two nodes `u` and `v`.
/// @attribute u  the initial node
/// @attribute v  the terminal node
.decl directed_edge(u: number, v: number)

directed_edge(1, 2).
directed_edge(1, 3).
directed_edge(2, 4).
directed_edge(3, 5).

/// There exists a path from node `u` to node `v`.
/// @attribute u  the source node
/// @attribute v  the sink node
.decl path(u: number, v: number)

path(u, v) :- directed_edge(u, v).
path(u, v) :- directed_edge(u, i), directed_edge(i, v)
