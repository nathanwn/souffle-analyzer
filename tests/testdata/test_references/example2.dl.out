example2.dl
 0 | /// There exists an edge between two nodes `u` and `v`.
 1 | /// @attribute u  the initial node
 2 | /// @attribute v  the terminal node
 3 | .decl directed_edge(u: number, v: number)
 4 |
 5 | directed_edge(1, 2).
 6 | directed_edge(1, 3).
 7 | directed_edge(2, 4).
 8 | directed_edge(3, 5).
 9 |
10 | /// There exists a path from node `u` to node `v`.
11 | /// @attribute u  the source node
12 | /// @attribute v  the sink node
13 | .decl path(u: number, v: number)
14 |
15 | path(u, v) :- directed_edge(u, v).
16 | path(u, v) :- directed_edge(u, i), directed_edge(i, v)

=====

-- Cursor range --
example2.dl
 3 | .decl directed_edge(u: number, v: number)
   |       ~~~~~~~~~~~~~
-- References --
example2.dl
 3 | .decl directed_edge(u: number, v: number)
   |       ~~~~~~~~~~~~~

example2.dl
 5 | directed_edge(1, 2).
   | ~~~~~~~~~~~~~

example2.dl
 6 | directed_edge(1, 3).
   | ~~~~~~~~~~~~~

example2.dl
 7 | directed_edge(2, 4).
   | ~~~~~~~~~~~~~

example2.dl
 8 | directed_edge(3, 5).
   | ~~~~~~~~~~~~~

example2.dl
15 | path(u, v) :- directed_edge(u, v).
   |               ~~~~~~~~~~~~~

example2.dl
16 | path(u, v) :- directed_edge(u, i), directed_edge(i, v)
   |               ~~~~~~~~~~~~~

example2.dl
16 | path(u, v) :- directed_edge(u, i), directed_edge(i, v)
   |                                    ~~~~~~~~~~~~~

_____

-- Cursor range --
example2.dl
 5 | directed_edge(1, 2).
   | ~~~~~~~~~~~~~
-- References --
example2.dl
 3 | .decl directed_edge(u: number, v: number)
   |       ~~~~~~~~~~~~~

example2.dl
 5 | directed_edge(1, 2).
   | ~~~~~~~~~~~~~

example2.dl
 6 | directed_edge(1, 3).
   | ~~~~~~~~~~~~~

example2.dl
 7 | directed_edge(2, 4).
   | ~~~~~~~~~~~~~

example2.dl
 8 | directed_edge(3, 5).
   | ~~~~~~~~~~~~~

example2.dl
15 | path(u, v) :- directed_edge(u, v).
   |               ~~~~~~~~~~~~~

example2.dl
16 | path(u, v) :- directed_edge(u, i), directed_edge(i, v)
   |               ~~~~~~~~~~~~~

example2.dl
16 | path(u, v) :- directed_edge(u, i), directed_edge(i, v)
   |                                    ~~~~~~~~~~~~~

_____

-- Cursor range --
example2.dl
 6 | directed_edge(1, 3).
   | ~~~~~~~~~~~~~
-- References --
example2.dl
 3 | .decl directed_edge(u: number, v: number)
   |       ~~~~~~~~~~~~~

example2.dl
 5 | directed_edge(1, 2).
   | ~~~~~~~~~~~~~

example2.dl
 6 | directed_edge(1, 3).
   | ~~~~~~~~~~~~~

example2.dl
 7 | directed_edge(2, 4).
   | ~~~~~~~~~~~~~

example2.dl
 8 | directed_edge(3, 5).
   | ~~~~~~~~~~~~~

example2.dl
15 | path(u, v) :- directed_edge(u, v).
   |               ~~~~~~~~~~~~~

example2.dl
16 | path(u, v) :- directed_edge(u, i), directed_edge(i, v)
   |               ~~~~~~~~~~~~~

example2.dl
16 | path(u, v) :- directed_edge(u, i), directed_edge(i, v)
   |                                    ~~~~~~~~~~~~~

_____

-- Cursor range --
example2.dl
 7 | directed_edge(2, 4).
   | ~~~~~~~~~~~~~
-- References --
example2.dl
 3 | .decl directed_edge(u: number, v: number)
   |       ~~~~~~~~~~~~~

example2.dl
 5 | directed_edge(1, 2).
   | ~~~~~~~~~~~~~

example2.dl
 6 | directed_edge(1, 3).
   | ~~~~~~~~~~~~~

example2.dl
 7 | directed_edge(2, 4).
   | ~~~~~~~~~~~~~

example2.dl
 8 | directed_edge(3, 5).
   | ~~~~~~~~~~~~~

example2.dl
15 | path(u, v) :- directed_edge(u, v).
   |               ~~~~~~~~~~~~~

example2.dl
16 | path(u, v) :- directed_edge(u, i), directed_edge(i, v)
   |               ~~~~~~~~~~~~~

example2.dl
16 | path(u, v) :- directed_edge(u, i), directed_edge(i, v)
   |                                    ~~~~~~~~~~~~~

_____

-- Cursor range --
example2.dl
 8 | directed_edge(3, 5).
   | ~~~~~~~~~~~~~
-- References --
example2.dl
 3 | .decl directed_edge(u: number, v: number)
   |       ~~~~~~~~~~~~~

example2.dl
 5 | directed_edge(1, 2).
   | ~~~~~~~~~~~~~

example2.dl
 6 | directed_edge(1, 3).
   | ~~~~~~~~~~~~~

example2.dl
 7 | directed_edge(2, 4).
   | ~~~~~~~~~~~~~

example2.dl
 8 | directed_edge(3, 5).
   | ~~~~~~~~~~~~~

example2.dl
15 | path(u, v) :- directed_edge(u, v).
   |               ~~~~~~~~~~~~~

example2.dl
16 | path(u, v) :- directed_edge(u, i), directed_edge(i, v)
   |               ~~~~~~~~~~~~~

example2.dl
16 | path(u, v) :- directed_edge(u, i), directed_edge(i, v)
   |                                    ~~~~~~~~~~~~~

_____

-- Cursor range --
example2.dl
13 | .decl path(u: number, v: number)
   |       ~~~~
-- References --
example2.dl
13 | .decl path(u: number, v: number)
   |       ~~~~

example2.dl
15 | path(u, v) :- directed_edge(u, v).
   | ~~~~

example2.dl
16 | path(u, v) :- directed_edge(u, i), directed_edge(i, v)
   | ~~~~

_____

-- Cursor range --
example2.dl
15 | path(u, v) :- directed_edge(u, v).
   | ~~~~
-- References --
example2.dl
13 | .decl path(u: number, v: number)
   |       ~~~~

example2.dl
15 | path(u, v) :- directed_edge(u, v).
   | ~~~~

example2.dl
16 | path(u, v) :- directed_edge(u, i), directed_edge(i, v)
   | ~~~~

_____

-- Cursor range --
example2.dl
15 | path(u, v) :- directed_edge(u, v).
   |               ~~~~~~~~~~~~~
-- References --
example2.dl
 3 | .decl directed_edge(u: number, v: number)
   |       ~~~~~~~~~~~~~

example2.dl
 5 | directed_edge(1, 2).
   | ~~~~~~~~~~~~~

example2.dl
 6 | directed_edge(1, 3).
   | ~~~~~~~~~~~~~

example2.dl
 7 | directed_edge(2, 4).
   | ~~~~~~~~~~~~~

example2.dl
 8 | directed_edge(3, 5).
   | ~~~~~~~~~~~~~

example2.dl
15 | path(u, v) :- directed_edge(u, v).
   |               ~~~~~~~~~~~~~

example2.dl
16 | path(u, v) :- directed_edge(u, i), directed_edge(i, v)
   |               ~~~~~~~~~~~~~

example2.dl
16 | path(u, v) :- directed_edge(u, i), directed_edge(i, v)
   |                                    ~~~~~~~~~~~~~

_____

-- Cursor range --
example2.dl
16 | path(u, v) :- directed_edge(u, i), directed_edge(i, v)
   | ~~~~
-- References --
example2.dl
13 | .decl path(u: number, v: number)
   |       ~~~~

example2.dl
15 | path(u, v) :- directed_edge(u, v).
   | ~~~~

example2.dl
16 | path(u, v) :- directed_edge(u, i), directed_edge(i, v)
   | ~~~~

_____

-- Cursor range --
example2.dl
16 | path(u, v) :- directed_edge(u, i), directed_edge(i, v)
   |               ~~~~~~~~~~~~~
-- References --
example2.dl
 3 | .decl directed_edge(u: number, v: number)
   |       ~~~~~~~~~~~~~

example2.dl
 5 | directed_edge(1, 2).
   | ~~~~~~~~~~~~~

example2.dl
 6 | directed_edge(1, 3).
   | ~~~~~~~~~~~~~

example2.dl
 7 | directed_edge(2, 4).
   | ~~~~~~~~~~~~~

example2.dl
 8 | directed_edge(3, 5).
   | ~~~~~~~~~~~~~

example2.dl
15 | path(u, v) :- directed_edge(u, v).
   |               ~~~~~~~~~~~~~

example2.dl
16 | path(u, v) :- directed_edge(u, i), directed_edge(i, v)
   |               ~~~~~~~~~~~~~

example2.dl
16 | path(u, v) :- directed_edge(u, i), directed_edge(i, v)
   |                                    ~~~~~~~~~~~~~

_____

-- Cursor range --
example2.dl
16 | path(u, v) :- directed_edge(u, i), directed_edge(i, v)
   |                                    ~~~~~~~~~~~~~
-- References --
example2.dl
 3 | .decl directed_edge(u: number, v: number)
   |       ~~~~~~~~~~~~~

example2.dl
 5 | directed_edge(1, 2).
   | ~~~~~~~~~~~~~

example2.dl
 6 | directed_edge(1, 3).
   | ~~~~~~~~~~~~~

example2.dl
 7 | directed_edge(2, 4).
   | ~~~~~~~~~~~~~

example2.dl
 8 | directed_edge(3, 5).
   | ~~~~~~~~~~~~~

example2.dl
15 | path(u, v) :- directed_edge(u, v).
   |               ~~~~~~~~~~~~~

example2.dl
16 | path(u, v) :- directed_edge(u, i), directed_edge(i, v)
   |               ~~~~~~~~~~~~~

example2.dl
16 | path(u, v) :- directed_edge(u, i), directed_edge(i, v)
   |                                    ~~~~~~~~~~~~~

_____
